client/landing/stepper/constants.ts

```diff
@@ -13,3 +13,20 @@ export const HOW_TO_MIGRATE_OPTIONS = {
 	DO_IT_FOR_ME: 'difm',
 	DO_IT_MYSELF: 'myself',
 };
+
+/**
+ * All Tracks events related to Stepper.
+ * Prefixed with `STEPPER_TRACKS_EVENT_[scope]_[action]` to avoid conflicts with other Tracks events.
+ * Example: `STEPPER_TRACKS_EVENT_STEP_NAV_SUBMIT` -> scope = `STEP_NAV`, action = `SUBMIT`
+ */
```

Comment:
@Automattic/dotcom-stepper I'd like us to have a system of constants and derived types for Tracks events handled by Stepper. What I'm proposing here is to start with the constant names (define a convention), reuse any existing event names for values, and adapt new events to match the naming convention of the constants. Gradually, the two (constants and their values) can sync.

Ideally, I'd like us to just straight-up create new events matching convention and abandon the existing ones (`calypso_signup_*`, etc.). Obviously, not as easy, but it's doable - people will learn to adapt their analyses.

I think Stepper should own its events. It's a system that's broader than signup and can pressumably be packaged and served in external environments (where "calypso" or "signup" aren't meaningful). So a good internal naming convention is key. Something like `stepper_flow_start`, `stepper_step_nav_next`, `stepper_step_nav_submit`, `stepper_flow_end`, etc.

WDYT?

Reply:
> Ideally, I'd like us to just straight-up create new events matching convention and abandon the existing ones (calypso_signup_*, etc.) from now. Obviously, not as easy, but it's doable - people will learn to adapt their analyses.

Well, actually - this isn't necessary. It's actually misleading. A consumer end can configure/define its own values, ideally. So in the constants defined here, `STEPPER_TRACKS_EVENT_STEP_NAV_SUBMIT` pointing to `calypso_signup_actions_submit_step` is akin to Calypso configuring the event to tailor to the Calyspo environment.

When/if Stepper is packaged, we can adopt a proper config system to do this properly.

Reply:
I'm in favor of new track events in Stepper, but we'll probably break many things if we get into that.

I like the constants idea. Seems like a good balance of having some convention without overhauling the events.

> When/if Stepper is packaged, we can adopt a proper config system to do this properly.

I don't think this will happen tbqh, nor that it should. I had the vision to not use Calypso state in Stepper when we started, but I now realize it was a mistake (more like a fools' errand). Calypso state is super powerful and versatile to move on from.

Reply:
> Calypso state is super powerful and versatile to move on from.

Hmm (especially) since the plans components migration to NPM we've actually needed a lot of the critical state be bundled in data-stores already - products, plans, purchases, sites, all have a data-store handle right now. I think the direction will be moving more and more slices there until there's an eventual single source. So that's happening already. Plans are pretty much pulling a lot of state out to serve outside of Calypso e.g. in logged-out/LandPack blocks in `/pricing`. 

I'm not sure if we have a choice really for a lot of these. Duplicated state handling isn't ideal, plus it's about 1/10th of the code when manage through React Query hooks.